core/src/main/java/org/elasticsearch/discovery/DiscoveryModule.java

```diff
@@ -62,16 +57,30 @@ public DiscoveryModule(Settings settings, TransportService transportService, Net
         addDiscoveryType("none", NoneDiscovery.class);
         addDiscoveryType("zen", ZenDiscovery.class);
 
-        Map<String, Supplier<UnicastHostsProvider>> hostProviders = new HashMap<>();
-        hostProviders.put("zen", () -> Collections::emptyList);
-        for (DiscoveryPlugin plugin : plugins) {
-            plugin.getZenHostsProviders(transportService, networkService).entrySet().forEach(entry -> {
-                if (hostProviders.put(entry.getKey(), entry.getValue()) != null) {
-                    throw new IllegalArgumentException("Cannot specify zen hosts provider [" + entry.getKey() + "] twice");
-                }
-            });
+        String discoveryType = DISCOVERY_TYPE_SETTING.get(settings);
+        if (discoveryType.equals("none") == false) {
+            Map<String, Supplier<UnicastHostsProvider>> hostProviders = new HashMap<>();
+            hostProviders.put("zen", () -> Collections::emptyList);
+            for (DiscoveryPlugin plugin : plugins) {
+                plugin.getZenHostsProviders(transportService, networkService).entrySet().forEach(entry -> {
```

Comment:
Also, part of my reasoning here for not doing a collection of pings is I don't think installing a plugin should automatically do something, I think it should be conscious choice by the user to say "I want to use this thing provided by the plugin I installed".